src/mono/mono/metadata/class-init.c

```diff
@@ -2999,6 +3033,293 @@ mono_class_setup_vtable_ginst (MonoClass *klass, GList *in_setup)
 
 }
 
```

Comment:
It might be a good idea to extract the vtable construction code into a separate file, its getting kinda big.